HowTos/Controller_Login_SAML_Config.rst

```diff
@@ -34,130 +36,227 @@ If you haven’t already deployed the Aviatrix controller, follow `the Controlle
+   |                         | such as "https://www.google.com"                |
+   +-------------------------+-------------------------------------------------+
+   | Entity ID               | Select `Hostname` for now                       |
+   +-------------------------+-------------------------------------------------+
+   | Access                  | Select admin or read-only access                |
+   +-------------------------+-------------------------------------------------+
+   | Custom SAML Request     | For now leave blank, depending on your specific |
+   | Template                | IdP, you may have to check this option          |
+   +-------------------------+-------------------------------------------------+
+
+.. note::
+   Each endpoint only supports one type of access. If you need admin and read-only access, create two separate SAML apps.
+
+#. Click `OK`
+#. Depending on your IdP provider, you may need to upload SP metadata. After temporary SAML endpoint is created:
+
+  - Right click **SP Metadata** button next to the SAML endpoint and save file to your local machine.
+  - Click **SP Metadata** button, and copy the SP metadata as text
+
+.. _Config_32:
+
+3.2 Create a SAML App for Aviatrix with the IdP
+###############################################
+
+.. note::
 
-This step is usually done by the IDP adminstrator.
+   This step is usually done by the IdP administrator.
+   This sections shows only a generalized process for creating a SAML application.
+   Refer to the `IdP-specific SAML App Integration <#idp-integration>`_ section for links to detailed steps with each particular IdP.
 
-Create a SAML 2.0 app with the following settings that are based on the <aviatrix_sp_name> which is 'controller' for controller login:
+Create a SAML 2.0 app with the IdP Provider. Enter the endpoint name you chose where it says <endpoint_name>
 
-#. Assertion Consumer Service URL* = https://aviatrix_controller_hostname/flask/saml/sso/<aviatrix_sp_name>
+#. Assertion Consumer Service URL* = https://aviatrix_controller_hostname/flask/saml/sso/<endpoint_name>
 #. Audience URI(Entity ID)* = https://aviatrix_controller_hostname/
-#. SP Metadata URL = https://aviatrix_controller_hostname/flask/saml/metadata/<aviatrix_sp_name>
-#. SP Login URL = https://aviatrix_controller_hostname/flask/saml/login/<aviatrix_sp_name>
+#. SP Metadata URL = https://aviatrix_controller_hostname/flask/metadata/<endpoint_name>
+#. SP Login URL = https://aviatrix_controller_hostname/flask/saml/login/<endpoint_name>
 #. Default RelayState* = <empty>
 
+|image3-2|
+
 .. important::
 
-   After step 3.3, these values are also available in the controller under the `Settings` navigation item.  Then, select `Controller` and go to the `SAML Login` tab.
+   After step 3.4, these values are also available in the controller under the `Settings` navigation item.  Then, select `Controller` and go to the `SAML Login` tab.
```

Comment:
At step 3.1 you already create a dummy endpoint. So say how to retrieve the values